src/inmanta/protocol/rest/server.py

```diff
@@ -81,19 +83,25 @@ def prepare(self) -> None:
             self.set_header("Access-Control-Allow-Origin", server_origin)
 
     def respond(self, body: ReturnTypes, headers: MutableMapping[str, str], status: int) -> None:
-        if CONTENT_TYPE not in headers:
-            headers[CONTENT_TYPE] = JSON_CONTENT
-        elif headers[CONTENT_TYPE] != JSON_CONTENT:
-            raise Exception("Invalid content type header provided. Only %s is supported" % JSON_CONTENT)
+        if common.CONTENT_TYPE not in headers:
+            headers[common.CONTENT_TYPE] = common.JSON_CONTENT
 
         if body is not None:
-            self.write(common.json_encode(body))
+            encoded_body = self._encode_body(body, headers[common.CONTENT_TYPE])
+            self.write(encoded_body)
 
         for header, value in headers.items():
             self.set_header(header, value)
 
         self.set_status(status)
 
+    def _encode_body(self, body: ReturnTypes, content_type: str) -> Union[ReturnTypes, bytes]:
+        if content_type == common.JSON_CONTENT:
+            return common.json_encode(body)
+        if content_type == common.HTML_CONTENT:
+            return body.encode(common.HTML_ENCODING)
```

Comment:
Why do we do this? This is indeed the default encoding but if the page defines it as utf-8 we might screw up the data by doing this. No?
cfr https://www.w3schools.com/tags/att_meta_charset.asp